doc/src/manual/parallel-computing.md

```diff
@@ -1,10 +1,265 @@
 # Parallel Computing
 
-This part of the manual details the following types of parallel programming available in Julia.
+For any newcomer to multi-thread and parallel computing it can be useful to first appreciate 
+the different levels of parallelism offered by Julia. We can divide them in three main categories :
+
+1. Green Threading or Julia Coroutines 
+2. Multi-Threading
+3. Multi-Core or Distributed Processing
+
+We will consider first Julia  [Tasks (aka Coroutines)](@ref man-tasks) and other high-level interfaces and 
+modules that rely on the Julia runtime library, that allow to suspend and resume computations with full 
+control of inter-`Tasks` coommunication without worrying of the opeartive system's scheduler. Julia also
+allows to communicate between `Tasks` through operations like [`wait`](@ref) and [`fetch`](@ref).
+
+Julia supports also multi-threading , where execution is forked and a anonymous function is run across all
+threads. Described as a fork-join approach, all threads have to join together and serial execution continues.
+Multi-threading is supported using the `Base.Threads` module that is still considered experimental, as Julia is
+not fully thread-safe, consider for example [this issue](https://github.com/JuliaLang/julia/issues/22581).
+Also multi-threading should be implemented considering properly global variables, locks and atomics, so we will
```

Comment:
considering properly -> properly considering?

Reply:
Ouch, yeah now I rephrased it a bit.